From 351232041fac161a50cc9c78a05773b4cfec13a5 Mon Sep 17 00:00:00 2001
Message-Id: <351232041fac161a50cc9c78a05773b4cfec13a5.1421709247.git.chang-joon.lee@intel.com>
In-Reply-To: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
References: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
From: Shobhit Kumar <shobhit.kumar@intel.com>
Date: Tue, 16 Sep 2014 18:52:25 +0530
Subject: [PATCH 23/75] MUST_REBASE [VPG]: adf/intel: Add REG_POSTING_READ
 Macro

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed

For: GMINL-3547
Change-Id: Ia4c4496a3d148dfdca34630e4d9033b7054177e2
Signed-off-by: Shobhit Kumar <shobhit.kumar@intel.com>
---
 drivers/video/adf/intel/include/intel_adf_device.h |    1 +
 drivers/video/adf/intel/intel_adf_device.c         |    5 +++++
 2 files changed, 6 insertions(+)

diff --git a/drivers/video/adf/intel/include/intel_adf_device.h b/drivers/video/adf/intel/include/intel_adf_device.h
index 0e49baf..9320951 100644
--- a/drivers/video/adf/intel/include/intel_adf_device.h
+++ b/drivers/video/adf/intel/include/intel_adf_device.h
@@ -64,6 +64,7 @@ extern struct intel_adf_device *intel_adf_device_create(struct pci_dev *pdev,
 extern void intel_adf_device_destroy(struct intel_adf_device *dev);
 
 extern u32 REG_READ(u32 reg);
+extern u32 REG_POSTING_READ(u32 reg);
 extern void REG_WRITE(u32 reg, u32 val);
 
 #endif /* INTEL_ADF_DEVICE_H_ */
diff --git a/drivers/video/adf/intel/intel_adf_device.c b/drivers/video/adf/intel/intel_adf_device.c
index 509c335..4884e64 100644
--- a/drivers/video/adf/intel/intel_adf_device.c
+++ b/drivers/video/adf/intel/intel_adf_device.c
@@ -39,6 +39,11 @@ void REG_WRITE(u32 reg, u32 val)
 	iowrite32(val, g_intel_adf_dev->mmio + reg);
 }
 
+u32 REG_POSTING_READ(u32 reg)
+{
+	return ioread32(g_intel_adf_dev->mmio + (reg));
+}
+
 struct post_obj {
 	void *obj;
 	struct list_head next;
-- 
1.7.9.5

